lib/embulk/guess/schema_guess.rb

```diff
@@ -62,11 +62,8 @@ def guess_type(str)
           return "long"
         end
 
-        if str.include?('.')
-          a, b = str.split(".", 2)
-          if a.to_i.to_s == a && b.to_i.to_s == b
-            return "double"
-          end
+        if str =~ /^[-+]?([1-9]\d*|0)(\.\d+)([eE][-+]?[0-9]+)?$/
```

Comment:
Got it. Then, it might be nice to have an explicit comment describing:
* It can be improved more later.
* It is intentionally rejecting "001" as a number.

I thought it's okay to "001" can be a number to be honest, but it should be determined by those who know the actual usage more.

To build a regex for that purpose, taking a look at [Guava's](https://github.com/google/guava/blob/4906fbbaeb22559563be02b0e765fda15426baff/guava/src/com/google/common/primitives/Doubles.java#L591-L599) may help instead of building from scratch. Dropping unnecessary parts from that may be easier.

Guava's license is Apache 2.0.